src/main/java/core/web/filters/AuthenticationFilter.java

```diff
@@ -0,0 +1,50 @@
+package core.web.filters;
+
+import core.lib.Injector;
+import core.service.driver.DriverService;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final Injector injector
+            = Injector.getInstance("core");
+    private static final String DRIVER_ID = "driverId";
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+    
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        throw new RuntimeException("Not supported");
+    }
+    
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+    
+        String url = req.getServletPath();
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (url.equals(req.getContextPath() + "/login")
+                || url.equals(req.getContextPath() + "/drivers/add")
+                || (driverId != null
+                && driverService.get(driverId) != null)) {
```

Comment:
```suggestion
        if (url.equals(req.getContextPath() + "/login")
                || url.equals(req.getContextPath() + "/drivers/add")
                || driverId != null) {
```
[5th](https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/09_filters_checklist.html) point in common mistakes

Reply:
Got it, Thanks